manifests/interface.pp

```diff
@@ -98,6 +102,10 @@
   Optional[String[1]] $private_key = undef,
   Optional[String[1]] $preshared_key = undef,
   Enum['systemd', 'wgquick'] $provider = 'systemd',
+  Optional[Array[String[1]]] $preup_cmds = undef,
```

Comment:
please default to an empty array instead. optional primitive datatypes should default to undef, scalar datatypes to their empty values (`[]` or `{}`).